src/containers/catalog/CatalogReducer.js

```diff
@@ -0,0 +1,47 @@
+/* @flow */
+import * as actionTypes from './CatalogActionTypes';
+
+export const INITIAL_STATE = {
+  asyncState: {
+    isLoading: true,
+    errorMessage: ""
+  },
+  entitySets: []
+};
+
+export function reducer(state = INITIAL_STATE, action) {
+  switch (action.type) {
+    case actionTypes.ENTITY_SET_LIST_REQUEST:
+      return Object.assign({}, state, {
+        asyncState: {
+          isLoading: true,
+          errorMessage: ""
+        },
+        entitySets: []
+      });
+      break;
+
+    case actionTypes.ENTITY_SET_LIST_FAILURE:
+      return Object.assign({}, state, {
+        asyncState: {
+          isLoading: false,
+          errorMessage: action.errorMessage
+        },
+        entitySets: []
+      });
+      break;
+
+    case actionTypes.ENTITY_SET_LIST_SUCCESS:
+      return Object.assign({}, state, {
```

Comment:
please use `immutable-js`. for example:

case ENTITY_SET_LIST_SUCCESS:
  return state
    .setIn(['asyncState', 'isLoading'], false)
    .setIn(['asyncState', 'errorMessage'], '')
    .set('entitySets', Immutable.fromJS(action.entitySets));